### PR TITLE
Toggle views

### PR DESCRIPTION
**Pull request by karmingc**
edited: may 22, 10:30pm 
- added auto suggest for job title 
- finalized UI design for grid view listing

changes:
toggle views = grid to list
- added modules for toggle views (VueX)
- added components for toggle views
- currently working auto suggest for location, job title, etc

### CODE REVIEW

File: src/components/MainPage/Listing.vue [hunk 1]
```diff
@@ -0,0 +1,68 @@
...
+        Grid,
+        List
+    },
+    // mounted() {
+    //     console.log('wasssup')
+    // },
+    computed: {
+        ...mapState({
+            // eslint-disable-next-line 
+            view: (state: any) => state.listings.view
```
Comment (CurtisMIT) [inline src/components/MainPage/Listing.vue]: Is it possible for you to create an interface for the state, this way we wouldn't need to disable eslint,
https://codeburst.io/vuex-and-typescript-3427ba78cfa8
this article demonstrates a nice way of doing so.

File: src/components/MainPage/ListingComponents/Grid.vue [hunk 2]
```diff
@@ -0,0 +1,133 @@
...
+
+<script lang="ts">
+import { Component, Vue } from 'vue-property-decorator';
+import { mapState } from 'vuex'
+
+@Component({        
+    computed: {
+        ...mapState({
+            // eslint-disable-next-line
+            entries: (state: any) => state.listings.entries
```
Comment (CurtisMIT) [inline src/components/MainPage/ListingComponents/Grid.vue]: Same here

**Review by CurtisMIT — CHANGES_REQUESTED** [review]
Integrate interfaces for the state, to avoid disabling es-lint in the future, everything else looks fine!

File: src/components/MainPage/ListingComponents/Grid.vue (hunk 2 above)
Comment (karmingc) [reply src/components/MainPage/ListingComponents/Grid.vue]: All changes have been made.
changes:
- interface for modules (actions, getters, states)
- removed eslint-disabled-next-line
- integrated vuex-class
plz npm install for new dependencies

**Comment by karmingc** [comment]
started individual page:
working on earnings: table completed, graph tomorrow

**Review by CurtisMIT — APPROVED** [review]

**Pull request merged**